src/Object/Post.php

```diff
@@ -172,13 +172,31 @@ public function getTemplateData($conv_responses, $thread_level = 1)
 			$dropping = true;
 		}
 
+		$origin = $item['origin'];
+
+		if (!$origin) {
+			/// @todo This shouldn't be done as query here, but better during the data creation.
+			// it is now done here, since during the RC phase we shouldn't make to intense changes.
+			$parent = dba::selectFirst('item', ['origin'], ['id' => $item['parent']]);
+			if (DBM::is_result($parent)) {
+				$origin = $parent['origin'];
+			}
+		}
+
+		// Showing the one or the other text, depending upon if we can only hide it or really delete it.
+		$delete = $origin ? L10n::t('Delete') : L10n::t('Remove from your stream');
```

Comment:
So, people will still be confused when they are allowed to delete something from their inbox I see m)

Reply:
Which inbox?

Reply:
When the system tells "delete" it **is** a deletion. Then it will be deleted not only for the user but fro everyone that subscribed to that thread. I think we have to tell the difference to the users.

Reply:
So if it is a public posting, possible one got over the relays, it is removed. But in private posting where I'm the only recipient on my server I can delete the posting? TBH it gets even more confusing.

The point in not calling it "delete" anymore was, because people get to scare by it hence we agreed on "remove from local timeline" and not call it hiding.

So your "delete" is now deletion and "remove from stream" is hiding as there are others who can still read it on your node?

Reply:
Could you please add the explanation to the relevant files in the documentation.

Reply:
It doesn't matter if it was a public or private post. It only matters if it was a post from us or a comment on one of our posts. Then they can be deleted - which will be federated across the network.

It is important that people know the difference. When - for example - people delete comments from others on their phone as some kind of "read" marker, then they have to know that this also deletes this comments from the systems of all followers on this post.

Reply:
Yes, please write it down in the documentation.

Reply:
Do you know a good place? I will try to do so. But at the moment I'm bug hunting in the pump.io connector.